src/platform/utilities/sso/index.js

```diff
@@ -53,16 +36,13 @@ export async function ssoKeepAliveSession() {
   return { ttl, authn };
 }
 
-export async function checkAutoSession() {
-  const [{ ttl, authn }, userProfile] = await Promise.all([
-    ssoKeepAliveSession(),
-    vaGovProfile(),
-  ]);
-  if (userProfile?.signIn?.ssoe) {
+export async function checkAutoSession(authenticatedWithSSOe) {
+  const { ttl, authn } = await ssoKeepAliveSession();
+
+  if (hasSession() && authenticatedWithSSOe) {
```

Comment:
This should be functional now. `hasSession` is present on page load, and since `AutoSSO` now makes sure the profile is done loading, we'll always know the status of authenticatedWithSSOe when this gets called. That has the added bonus that `checkAutoSession` should only ever get called once per page load, as was originally intended.

Reply:
We could probably decouple from `localStorage` here altogether and just use the `isLoggedIn` selector. Since this function isn't called if until the profile is done loading, that's just as reliable a value as `hasSession()`

Reply:
The part that I'm not tracking is that hasSession is derived from local storage - so great, whenever a page loads we'll have that present. And a lot of times, the user.profile will be loaded by an API call to `/v0/user` (in response to a sign-in or in other places where profile is used). But what if there's a page refresh and nothing calls `/v0/user`? 

If hasSession is true, then we need to know authenticatedWithSSOe to know how to proceed (the first outer branch of this method)
If hasSession is false, then we actually don't care, and can potentially proceed to auto login. So maybe AutoSSO should pass "whatever info it has" and then call this again as state updates with new info from profile? And if hasSession is true but user.profile is still loading, it just no-ops?

Reply:
Was writing in haste there, I guess it's two separate questions:
- Are there code paths where hasSession = true but we might be waiting for the user profile forever but nothing has made that API call?

- Does it make sense for this method to potentially be called multiple times as state changes in AutoSSO?

Reply:
Answering the two questions first and then a longer explanation:
* I don't think so, since hasSession = false also gives us enough information to act
* We could, but ultimately I think it might just result in an extra call to `keepAlive` before we have all the required info on the `vets-website` side to determine any inbound actions required

### Longer explanation

Within the user nav, which is site-wide (minus a few edge cases like standalone sign-in where we address it), this is the check happens right when the component mounts:
```js
checkLoggedInStatus = () => {
    if (hasSession()) {
      this.props.initializeProfile();
    } else {
      this.props.updateLoggedInStatus(false);
      if (this.getNextParameter()) this.openLoginModal();
    }
  };
```
`initializeProfile` is what tries to call `/v0/user` and then also calls `updateLoggedInStatus` accordingly.

Regardless of whether `hasSession` is true or not, the logic in `AutoSSO` now doesn't call `checkAutoSession` until `!profileLoading` evaluates to true, and _that_ doesn't happen until the above code block evaluates. By that point, API call or not, I believe it tells us everything we need to know for triggering any inbound responses (calling `updateLoggedInStatus` results in loading to be set to false).

Reply:
Got it, yes, that call to `initializeProfile` was what I was missing